R/labels.r

```diff
@@ -51,10 +61,17 @@ update_labels <- function(p, labels) {
 #' # The plot tag appears at the top-left, and is typically used
 #' # for labelling a subplot with a letter.
 #' p + labs(title = "title", tag = "A")
-labs <- function(...) {
-  args <- list(...)
-  if (is.list(args[[1]])) args <- args[[1]]
+#'
+#' # If you want to remove a label, set it to NULL.
+#' p + labs(title = "title") + labs(title = NULL)
+labs <- function(..., title = waiver(), subtitle = waiver(), caption = waiver(), tag = waiver()) {
+  args <- list(..., title = title, subtitle = subtitle, caption = caption, tag = tag)
+  if (length(args) > 0 && is.list(args[[1]]) && !is.waive(args[[1]])) args <- args[[1]]
```

Comment:
By the way, why should `labs()` accept a list as its first argument?
I couldn't find this is documented or used elsewhere.

Reply:
Now that ggplot2 uses tidy eval, I think we should remove this, and in the previous line use `rlang::list2()` instead of `list()`. That way you can use `!!!` if you do already have a list of labels.